sdk/trace/provider.go

```diff
@@ -366,6 +366,7 @@ func ensureValidTracerProviderConfig(cfg tracerProviderConfig) tracerProviderCon
 	if cfg.idGenerator == nil {
 		cfg.idGenerator = defaultIDGenerator()
 	}
+	cfg.spanLimits.parsePotentialEnvConfigs()
```

Comment:
This will overwrite the values set by `WithSpanLimits` if passed. This doesn't match our existing behavior: local configuration overrides environment configuration [^1][^2][^3][^4][^5]. [This is not well defined in the specification](https://github.com/open-telemetry/opentelemetry-specification/issues/1171), but I think we need to remain consistent in how we apply our configuration. Please apply the environment configuration prior to applying the passed local configuration options.

[^1]: https://github.com/open-telemetry/opentelemetry-go/blob/cd21df42c41f1ac22c80ffc93d877179b3fee98a/sdk/trace/batch_span_processor.go#L105
[^2]: https://github.com/open-telemetry/opentelemetry-go/blob/cd21df42c41f1ac22c80ffc93d877179b3fee98a/exporters/otlp/otlptrace/internal/otlpconfig/options.go#L86
[^3]: https://github.com/open-telemetry/opentelemetry-go/blob/cd21df42c41f1ac22c80ffc93d877179b3fee98a/exporters/otlp/otlpmetric/internal/otlpconfig/options.go#L93
[^4]: https://github.com/open-telemetry/opentelemetry-go/blob/cd21df42c41f1ac22c80ffc93d877179b3fee98a/exporters/zipkin/zipkin.go#L87
[^5]: https://github.com/open-telemetry/opentelemetry-go/blob/cd21df42c41f1ac22c80ffc93d877179b3fee98a/exporters/jaeger/uploader.go#L58-L67

Reply:
Done